dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/PeriodResourceTable.java

```diff
@@ -140,4 +142,20 @@ public String getCreateTempTableStatement()
 
         return Lists.newArrayList( sql );
     }
+    
+    private int resolveYearFromPeriod( Period period )
+    {
+        // Weekly type has to be treated separately from other Period types.
+        // In order to handle all weekly types uniformly, 3 days are added to the week start day and
+        // the year of the modified start date is used as reference year for the Period
+
+        if ( WeeklyAbstractPeriodType.class.isAssignableFrom( period.getPeriodType().getClass() ) )
+        {
+            return new DateTime( period.getStartDate().getTime() ).plusDays( 3 ).getYear();
```

Comment:
Looks great.

My only concern here is our famous Calendar system. I think we need to retrieve the current calendar and use that? We can check with Morten H.

Reply:
Ok, thanks, I'll add Morten to the reviewers

Reply:
When it comes to weeks we currently don't do much considerations for other calendars, mostly because they normally don't have properly defined weeks etc... so i think this is fine (for now at least lets just do ISO weeks for all calendars)

Reply:
Just to be clear, if you want to add support you can simple get the calendar, check if iso (then do what you did), if not.. you can get year from the the calendar instance, but generally we have not been considering weeks for other calendars (we might want to do better, but its hard when week periods are not defined in their calendar)

@abyot might have something to add about the Ethiopian calendar